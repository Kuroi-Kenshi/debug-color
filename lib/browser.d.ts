/**
 * Created by user on 2018/7/2/002.
 */
import ConsoleNode from './node';
export * from './node';
import { IOptions } from './val';
export declare class Console2 extends ConsoleNode {
    constructor(target?: Console, options?: IOptions);
}
export default Console2;
