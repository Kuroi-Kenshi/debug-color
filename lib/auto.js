"use strict";
function __export(m) {
    for (var p in m) if (!exports.hasOwnProperty(p)) exports[p] = m[p];
}
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
const chk_1 = __importDefault(require("./chk"));
__export(require("./node"));
let self;
if (chk_1.default()) {
    self = require('./node');
}
else {
    self = require('./browser');
}
module.exports = self;
//# sourceMappingURL=data:application/json;base64,eyJ2ZXJzaW9uIjozLCJmaWxlIjoiYXV0by5qcyIsInNvdXJjZVJvb3QiOiIiLCJzb3VyY2VzIjpbImF1dG8udHMiXSwibmFtZXMiOltdLCJtYXBwaW5ncyI6Ijs7Ozs7OztBQUNBLGdEQUE2QjtBQUM3Qiw0QkFBdUI7QUFFdkIsSUFBSSxJQUFJLENBQUM7QUFFVCxJQUFJLGFBQVEsRUFBRSxFQUNkO0lBQ0MsSUFBSSxHQUFHLE9BQU8sQ0FBQyxRQUFRLENBQUMsQ0FBQztDQUN6QjtLQUVEO0lBQ0MsSUFBSSxHQUFHLE9BQU8sQ0FBQyxXQUFXLENBQUMsQ0FBQztDQUM1QjtBQUdELGlCQUFTLElBQXNDLENBQUMiLCJzb3VyY2VzQ29udGVudCI6WyJcbmltcG9ydCBpc05vZGVKcyBmcm9tICcuL2Noayc7XG5leHBvcnQgKiBmcm9tICcuL25vZGUnO1xuXG5sZXQgc2VsZjtcblxuaWYgKGlzTm9kZUpzKCkpXG57XG5cdHNlbGYgPSByZXF1aXJlKCcuL25vZGUnKTtcbn1cbmVsc2Vcbntcblx0c2VsZiA9IHJlcXVpcmUoJy4vYnJvd3NlcicpO1xufVxuXG4vLyBAdHMtaWdub3JlXG5leHBvcnQgPSBzZWxmIGFzIGFueSBhcyB0eXBlb2YgaW1wb3J0KCcuL25vZGUnKTtcbiJdfQ==