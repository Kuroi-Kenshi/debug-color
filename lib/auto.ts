
import isNodeJs from './chk';
import * as CM from './node';
export * from './node';

let self;

if (isNodeJs())
{
	self = require('./node');
}
else
{
	self = require('./browser');
}

exports = self;
