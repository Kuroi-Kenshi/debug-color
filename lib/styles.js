"use strict";
/**
 * Created by user on 2018/6/26/026.
 */
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const chalk_1 = __importDefault(require("chalk"));
const prototype = chalk_1.default.constructor.prototype;
exports.styleNames = Object
    .getOwnPropertyNames(prototype)
    .filter(v => v != 'constructor');
exports.styleNamesFn = [
    'rgb',
    'hsl',
    'hsv',
    'hwb',
    'bgHex',
    'bgKeyword',
    'bgRgb',
    'bgHsl',
    'bgHsv',
    'bgHwb',
    'hex',
    'keyword',
];
exports.default = exports;
//# sourceMappingURL=data:application/json;base64,eyJ2ZXJzaW9uIjozLCJmaWxlIjoic3R5bGVzLmpzIiwic291cmNlUm9vdCI6IiIsInNvdXJjZXMiOlsic3R5bGVzLnRzIl0sIm5hbWVzIjpbXSwibWFwcGluZ3MiOiI7QUFBQTs7R0FFRzs7Ozs7QUFFSCxrREFBMEI7QUFFMUIsTUFBTSxTQUFTLEdBQUcsZUFBSyxDQUFDLFdBQVcsQ0FBQyxTQUFTLENBQUM7QUFFakMsUUFBQSxVQUFVLEdBQUcsTUFBTTtLQUM5QixtQkFBbUIsQ0FBQyxTQUFTLENBQUM7S0FDOUIsTUFBTSxDQUFDLENBQUMsQ0FBQyxFQUFFLENBQUMsQ0FBQyxJQUFJLGFBQWEsQ0FBc0IsQ0FDckQ7QUFFWSxRQUFBLFlBQVksR0FBcUI7SUFDN0MsS0FBSztJQUNMLEtBQUs7SUFDTCxLQUFLO0lBQ0wsS0FBSztJQUNMLE9BQU87SUFDUCxXQUFXO0lBQ1gsT0FBTztJQUNQLE9BQU87SUFDUCxPQUFPO0lBQ1AsT0FBTztJQUNQLEtBQUs7SUFDTCxTQUFTO0NBQ1QsQ0FBQztBQTZGRixrQkFBZSxPQUFvQyxDQUFBIiwic291cmNlc0NvbnRlbnQiOlsiLyoqXG4gKiBDcmVhdGVkIGJ5IHVzZXIgb24gMjAxOC82LzI2LzAyNi5cbiAqL1xuXG5pbXBvcnQgY2hhbGsgZnJvbSAnY2hhbGsnO1xuXG5jb25zdCBwcm90b3R5cGUgPSBjaGFsay5jb25zdHJ1Y3Rvci5wcm90b3R5cGU7XG5cbmV4cG9ydCBjb25zdCBzdHlsZU5hbWVzID0gT2JqZWN0XG5cdC5nZXRPd25Qcm9wZXJ0eU5hbWVzKHByb3RvdHlwZSlcblx0LmZpbHRlcih2ID0+IHYgIT0gJ2NvbnN0cnVjdG9yJykgYXMgKGtleW9mIElTdHlsZXMpW11cbjtcblxuZXhwb3J0IGNvbnN0IHN0eWxlTmFtZXNGbjogSVN0eWxlc0ZuTmFtZXNbXSA9IFtcblx0J3JnYicsXG5cdCdoc2wnLFxuXHQnaHN2Jyxcblx0J2h3YicsXG5cdCdiZ0hleCcsXG5cdCdiZ0tleXdvcmQnLFxuXHQnYmdSZ2InLFxuXHQnYmdIc2wnLFxuXHQnYmdIc3YnLFxuXHQnYmdId2InLFxuXHQnaGV4Jyxcblx0J2tleXdvcmQnLFxuXTtcblxuZXhwb3J0IHR5cGUgSVN0eWxlc0ZuTmFtZXMgPVxuXHQncmdiJ1xuXHR8ICdoc2wnXG5cdHwgJ2hzdidcblx0fCAnaHdiJ1xuXHR8ICdiZ0hleCdcblx0fCAnYmdLZXl3b3JkJ1xuXHR8ICdiZ1JnYidcblx0fCAnYmdIc2wnXG5cdHwgJ2JnSHN2J1xuXHR8ICdiZ0h3Yidcblx0fCAnaGV4J1xuXHR8ICdrZXl3b3JkJ1xuXHQ7XG5cbmV4cG9ydCB0eXBlIElTdHlsZXNDb2xvck5hbWVzID0gRXhjbHVkZTxrZXlvZiBJU3R5bGVzLCBJU3R5bGVzRm5OYW1lcz47XG5cbmV4cG9ydCBpbnRlcmZhY2UgSVN0eWxlc1xue1xuXHRyZ2IocjogbnVtYmVyLCBnOiBudW1iZXIsIGI6IG51bWJlcik6IHRoaXM7XG5cblx0aHNsKGg6IG51bWJlciwgczogbnVtYmVyLCBsOiBudW1iZXIpOiB0aGlzO1xuXG5cdGhzdihoOiBudW1iZXIsIHM6IG51bWJlciwgdjogbnVtYmVyKTogdGhpcztcblxuXHRod2IoaDogbnVtYmVyLCB3OiBudW1iZXIsIGI6IG51bWJlcik6IHRoaXM7XG5cblx0YmdIZXgoY29sb3I6IHN0cmluZyk6IHRoaXM7XG5cblx0YmdLZXl3b3JkKGNvbG9yOiBzdHJpbmcpOiB0aGlzO1xuXG5cdGJnUmdiKHI6IG51bWJlciwgZzogbnVtYmVyLCBiOiBudW1iZXIpOiB0aGlzO1xuXG5cdGJnSHNsKGg6IG51bWJlciwgczogbnVtYmVyLCBsOiBudW1iZXIpOiB0aGlzO1xuXG5cdGJnSHN2KGg6IG51bWJlciwgczogbnVtYmVyLCB2OiBudW1iZXIpOiB0aGlzO1xuXG5cdGJnSHdiKGg6IG51bWJlciwgdzogbnVtYmVyLCBiOiBudW1iZXIpOiB0aGlzO1xuXG5cdGhleChjb2xvcjogc3RyaW5nKTogdGhpcztcblxuXHRrZXl3b3JkKGNvbG9yOiBzdHJpbmcpOiB0aGlzO1xuXG5cdHJlYWRvbmx5IHJlc2V0OiB0aGlzO1xuXHRyZWFkb25seSBib2xkOiB0aGlzO1xuXHRyZWFkb25seSBkaW06IHRoaXM7XG5cdHJlYWRvbmx5IGl0YWxpYzogdGhpcztcblx0cmVhZG9ubHkgdW5kZXJsaW5lOiB0aGlzO1xuXHRyZWFkb25seSBpbnZlcnNlOiB0aGlzO1xuXHRyZWFkb25seSBoaWRkZW46IHRoaXM7XG5cdHJlYWRvbmx5IHN0cmlrZXRocm91Z2g6IHRoaXM7XG5cblx0cmVhZG9ubHkgdmlzaWJsZTogdGhpcztcblxuXHRyZWFkb25seSBibGFjazogdGhpcztcblx0cmVhZG9ubHkgcmVkOiB0aGlzO1xuXHRyZWFkb25seSBncmVlbjogdGhpcztcblx0cmVhZG9ubHkgeWVsbG93OiB0aGlzO1xuXHRyZWFkb25seSBibHVlOiB0aGlzO1xuXHRyZWFkb25seSBtYWdlbnRhOiB0aGlzO1xuXHRyZWFkb25seSBjeWFuOiB0aGlzO1xuXHRyZWFkb25seSB3aGl0ZTogdGhpcztcblx0cmVhZG9ubHkgZ3JheTogdGhpcztcblx0cmVhZG9ubHkgZ3JleTogdGhpcztcblx0cmVhZG9ubHkgYmxhY2tCcmlnaHQ6IHRoaXM7XG5cdHJlYWRvbmx5IHJlZEJyaWdodDogdGhpcztcblx0cmVhZG9ubHkgZ3JlZW5CcmlnaHQ6IHRoaXM7XG5cdHJlYWRvbmx5IHllbGxvd0JyaWdodDogdGhpcztcblx0cmVhZG9ubHkgYmx1ZUJyaWdodDogdGhpcztcblx0cmVhZG9ubHkgbWFnZW50YUJyaWdodDogdGhpcztcblx0cmVhZG9ubHkgY3lhbkJyaWdodDogdGhpcztcblx0cmVhZG9ubHkgd2hpdGVCcmlnaHQ6IHRoaXM7XG5cblx0cmVhZG9ubHkgYmdCbGFjazogdGhpcztcblx0cmVhZG9ubHkgYmdSZWQ6IHRoaXM7XG5cdHJlYWRvbmx5IGJnR3JlZW46IHRoaXM7XG5cdHJlYWRvbmx5IGJnWWVsbG93OiB0aGlzO1xuXHRyZWFkb25seSBiZ0JsdWU6IHRoaXM7XG5cdHJlYWRvbmx5IGJnTWFnZW50YTogdGhpcztcblx0cmVhZG9ubHkgYmdDeWFuOiB0aGlzO1xuXHRyZWFkb25seSBiZ1doaXRlOiB0aGlzO1xuXHRyZWFkb25seSBiZ0JsYWNrQnJpZ2h0OiB0aGlzO1xuXHRyZWFkb25seSBiZ1JlZEJyaWdodDogdGhpcztcblx0cmVhZG9ubHkgYmdHcmVlbkJyaWdodDogdGhpcztcblx0cmVhZG9ubHkgYmdZZWxsb3dCcmlnaHQ6IHRoaXM7XG5cdHJlYWRvbmx5IGJnQmx1ZUJyaWdodDogdGhpcztcblx0cmVhZG9ubHkgYmdNYWdlbnRhQnJpZ2h0OiB0aGlzO1xuXHRyZWFkb25seSBiZ0N5YW5CcmlnaHQ6IHRoaXM7XG5cdHJlYWRvbmx5IGJnV2hpdGVCcmlnaHQ6IHRoaXM7XG59XG5cbmV4cG9ydCBkZWZhdWx0IGV4cG9ydHMgYXMgdHlwZW9mIGltcG9ydCgnLi9zdHlsZXMnKVxuIl19