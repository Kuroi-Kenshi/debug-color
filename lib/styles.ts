/**
 * Created by user on 2018/6/26/026.
 */

import chalk from 'chalk';
import { IStyles, IStylesFnNames, IStylesNamesWithoutFn } from './types';

export * from './types';

export function getStyleNamesByChalk(chalk): (keyof IStyles)[]
{
	const prototype = chalk.constructor.prototype;

	return Object
		.getOwnPropertyNames(prototype)
		.filter(v => v != 'constructor') as (keyof IStyles)[]
		;
}

export const styleNames = getStyleNamesByChalk(chalk);

export const styleNamesFn: IStylesFnNames[] = [
	'rgb',
	'hsl',
	'hsv',
	'hwb',
	'bgHex',
	'bgKeyword',
	'bgRgb',
	'bgHsl',
	'bgHsv',
	'bgHwb',
	'hex',
	'keyword',
];

export const styleNamesWithoutFn = styleNames.filter(name => !styleNamesFn.includes(name as any)) as IStylesNamesWithoutFn

export default exports as typeof import('./styles')
