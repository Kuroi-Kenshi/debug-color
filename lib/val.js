"use strict";
/**
 * Created by user on 2018/6/26/026.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SYM_DEBUG_CONSOLE = Symbol('DebugConsole');
exports.SYM_CHALK = Symbol('chalk');
exports.SYM_CONSOLE = Symbol('console');
exports.SYM_EVENT = Symbol('event');
exports.SYM_DATA = Symbol('data');
const val = require("./val");
exports.default = val;
