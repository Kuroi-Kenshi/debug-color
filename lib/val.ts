/**
 * Created by user on 2018/6/26/026.
 */

import { Chalk } from 'chalk';
import * as util from 'util';
import * as val from './val';
export { InspectOptions } from 'util';

export const SYM_DEBUG_CONSOLE = Symbol('DebugConsole');
export const SYM_CHALK = Symbol('chalk');

export const SYM_CONSOLE = Symbol('console');

export const SYM_EVENT = Symbol('event');

export const SYM_DATA = Symbol('data');

export default val;

export interface IOptions
{
	/**
	 * enable log display or not
	 */
	enabled?: boolean,

	/**
	 * show label string
	 */
	label?: boolean | string[],

	/**
	 * show time label
	 */
	time?: boolean,

	/**
	 * set default inspectOptions
	 */
	inspectOptions?: util.InspectOptions,

	/**
	 * set color style
	 */
	colors?: {
		debug?,
		error?,
		info?,
		log?,
		trace?,
		warn?,
		success?,
		ok?,
		[k: string]: string | Chalk | IChalkLike,
	},

	/**
	 * check is node.js console
	 */
	readonly stream?: boolean,
}

export const defaultColors: IOptions["colors"] = {
	error: 'red',
	exception: 'red',
	warn: 'red',

	fail: 'red',

	info: 'cyan',
	debug: 'cyan',

	success: 'green',
	ok: 'green',
};

export interface IChalkLike
{
	(string): string,

	(string, ...argv): string,

	(...argv): string,
}
