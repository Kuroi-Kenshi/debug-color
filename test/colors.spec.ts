import { styleNames, styleNamesFn, IStylesNameWithoutFn, styleNamesWithoutFn } from '../lib/styles';
import { console, Console2 } from '..';
import { Console } from 'console';
import { SYM_CONSOLE, SYM_CHALK, SYM_DATA } from '../lib/val';

describe(`styleNamesWithoutFn`, () =>
{

	styleNamesWithoutFn
		.forEach((color) =>
		{
			test(color, () =>
			{

				let actual = console[color];

				expect(typeof actual).toMatchSnapshot();

				const data = {
					[SYM_CHALK]: actual[SYM_CHALK],
					[SYM_DATA]: actual[SYM_DATA],
				}

				expect(actual[SYM_CONSOLE]).toBeInstanceOf(Console);

				expect(actual.log).toBeInstanceOf(Function);
				expect(actual.dir).toBeInstanceOf(Function);
				expect(actual.error).toBeInstanceOf(Function);
				expect(actual.warn).toBeInstanceOf(Function);
				expect(actual.time).toBeInstanceOf(Function);
				expect(actual.timeEnd).toBeInstanceOf(Function);
				expect(actual.debug).toBeInstanceOf(Function);

				expect(data).toMatchSnapshot();

			});
		})
	;

})
