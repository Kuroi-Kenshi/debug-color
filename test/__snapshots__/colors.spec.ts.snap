// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`styleNamesWithoutFn ansi 1`] = `"function"`;

exports[`styleNamesWithoutFn ansi 2`] = `
Object {
  Symbol(chalk): [Function],
  Symbol(data): Object {
    "chalkOptions": Object {},
    "stream": true,
  },
}
`;

exports[`styleNamesWithoutFn ansi256 1`] = `"function"`;

exports[`styleNamesWithoutFn ansi256 2`] = `
Object {
  Symbol(chalk): [Function],
  Symbol(data): Object {
    "chalkOptions": Object {},
    "stream": true,
  },
}
`;

exports[`styleNamesWithoutFn apple 1`] = `"function"`;

exports[`styleNamesWithoutFn apple 2`] = `
Object {
  Symbol(chalk): [Function],
  Symbol(data): Object {
    "chalkOptions": Object {},
    "stream": true,
  },
}
`;

exports[`styleNamesWithoutFn bgAnsi 1`] = `"function"`;

exports[`styleNamesWithoutFn bgAnsi 2`] = `
Object {
  Symbol(chalk): [Function],
  Symbol(data): Object {
    "chalkOptions": Object {},
    "stream": true,
  },
}
`;

exports[`styleNamesWithoutFn bgAnsi256 1`] = `"function"`;

exports[`styleNamesWithoutFn bgAnsi256 2`] = `
Object {
  Symbol(chalk): [Function],
  Symbol(data): Object {
    "chalkOptions": Object {},
    "stream": true,
  },
}
`;

exports[`styleNamesWithoutFn bgApple 1`] = `"function"`;

exports[`styleNamesWithoutFn bgApple 2`] = `
Object {
  Symbol(chalk): [Function],
  Symbol(data): Object {
    "chalkOptions": Object {},
    "stream": true,
  },
}
`;

exports[`styleNamesWithoutFn bgBlack 1`] = `"function"`;

exports[`styleNamesWithoutFn bgBlack 2`] = `
Object {
  Symbol(chalk): [Function],
  Symbol(data): Object {
    "chalkOptions": Object {},
    "stream": true,
  },
}
`;

exports[`styleNamesWithoutFn bgBlackBright 1`] = `"function"`;

exports[`styleNamesWithoutFn bgBlackBright 2`] = `
Object {
  Symbol(chalk): [Function],
  Symbol(data): Object {
    "chalkOptions": Object {},
    "stream": true,
  },
}
`;

exports[`styleNamesWithoutFn bgBlue 1`] = `"function"`;

exports[`styleNamesWithoutFn bgBlue 2`] = `
Object {
  Symbol(chalk): [Function],
  Symbol(data): Object {
    "chalkOptions": Object {},
    "stream": true,
  },
}
`;

exports[`styleNamesWithoutFn bgBlueBright 1`] = `"function"`;

exports[`styleNamesWithoutFn bgBlueBright 2`] = `
Object {
  Symbol(chalk): [Function],
  Symbol(data): Object {
    "chalkOptions": Object {},
    "stream": true,
  },
}
`;

exports[`styleNamesWithoutFn bgCmyk 1`] = `"function"`;

exports[`styleNamesWithoutFn bgCmyk 2`] = `
Object {
  Symbol(chalk): [Function],
  Symbol(data): Object {
    "chalkOptions": Object {},
    "stream": true,
  },
}
`;

exports[`styleNamesWithoutFn bgCyan 1`] = `"function"`;

exports[`styleNamesWithoutFn bgCyan 2`] = `
Object {
  Symbol(chalk): [Function],
  Symbol(data): Object {
    "chalkOptions": Object {},
    "stream": true,
  },
}
`;

exports[`styleNamesWithoutFn bgCyanBright 1`] = `"function"`;

exports[`styleNamesWithoutFn bgCyanBright 2`] = `
Object {
  Symbol(chalk): [Function],
  Symbol(data): Object {
    "chalkOptions": Object {},
    "stream": true,
  },
}
`;

exports[`styleNamesWithoutFn bgGreen 1`] = `"function"`;

exports[`styleNamesWithoutFn bgGreen 2`] = `
Object {
  Symbol(chalk): [Function],
  Symbol(data): Object {
    "chalkOptions": Object {},
    "stream": true,
  },
}
`;

exports[`styleNamesWithoutFn bgGreenBright 1`] = `"function"`;

exports[`styleNamesWithoutFn bgGreenBright 2`] = `
Object {
  Symbol(chalk): [Function],
  Symbol(data): Object {
    "chalkOptions": Object {},
    "stream": true,
  },
}
`;

exports[`styleNamesWithoutFn bgHcg 1`] = `"function"`;

exports[`styleNamesWithoutFn bgHcg 2`] = `
Object {
  Symbol(chalk): [Function],
  Symbol(data): Object {
    "chalkOptions": Object {},
    "stream": true,
  },
}
`;

exports[`styleNamesWithoutFn bgLab 1`] = `"function"`;

exports[`styleNamesWithoutFn bgLab 2`] = `
Object {
  Symbol(chalk): [Function],
  Symbol(data): Object {
    "chalkOptions": Object {},
    "stream": true,
  },
}
`;

exports[`styleNamesWithoutFn bgLch 1`] = `"function"`;

exports[`styleNamesWithoutFn bgLch 2`] = `
Object {
  Symbol(chalk): [Function],
  Symbol(data): Object {
    "chalkOptions": Object {},
    "stream": true,
  },
}
`;

exports[`styleNamesWithoutFn bgMagenta 1`] = `"function"`;

exports[`styleNamesWithoutFn bgMagenta 2`] = `
Object {
  Symbol(chalk): [Function],
  Symbol(data): Object {
    "chalkOptions": Object {},
    "stream": true,
  },
}
`;

exports[`styleNamesWithoutFn bgMagentaBright 1`] = `"function"`;

exports[`styleNamesWithoutFn bgMagentaBright 2`] = `
Object {
  Symbol(chalk): [Function],
  Symbol(data): Object {
    "chalkOptions": Object {},
    "stream": true,
  },
}
`;

exports[`styleNamesWithoutFn bgRed 1`] = `"function"`;

exports[`styleNamesWithoutFn bgRed 2`] = `
Object {
  Symbol(chalk): [Function],
  Symbol(data): Object {
    "chalkOptions": Object {},
    "stream": true,
  },
}
`;

exports[`styleNamesWithoutFn bgRedBright 1`] = `"function"`;

exports[`styleNamesWithoutFn bgRedBright 2`] = `
Object {
  Symbol(chalk): [Function],
  Symbol(data): Object {
    "chalkOptions": Object {},
    "stream": true,
  },
}
`;

exports[`styleNamesWithoutFn bgWhite 1`] = `"function"`;

exports[`styleNamesWithoutFn bgWhite 2`] = `
Object {
  Symbol(chalk): [Function],
  Symbol(data): Object {
    "chalkOptions": Object {},
    "stream": true,
  },
}
`;

exports[`styleNamesWithoutFn bgWhiteBright 1`] = `"function"`;

exports[`styleNamesWithoutFn bgWhiteBright 2`] = `
Object {
  Symbol(chalk): [Function],
  Symbol(data): Object {
    "chalkOptions": Object {},
    "stream": true,
  },
}
`;

exports[`styleNamesWithoutFn bgXyz 1`] = `"function"`;

exports[`styleNamesWithoutFn bgXyz 2`] = `
Object {
  Symbol(chalk): [Function],
  Symbol(data): Object {
    "chalkOptions": Object {},
    "stream": true,
  },
}
`;

exports[`styleNamesWithoutFn bgYellow 1`] = `"function"`;

exports[`styleNamesWithoutFn bgYellow 2`] = `
Object {
  Symbol(chalk): [Function],
  Symbol(data): Object {
    "chalkOptions": Object {},
    "stream": true,
  },
}
`;

exports[`styleNamesWithoutFn bgYellowBright 1`] = `"function"`;

exports[`styleNamesWithoutFn bgYellowBright 2`] = `
Object {
  Symbol(chalk): [Function],
  Symbol(data): Object {
    "chalkOptions": Object {},
    "stream": true,
  },
}
`;

exports[`styleNamesWithoutFn black 1`] = `"function"`;

exports[`styleNamesWithoutFn black 2`] = `
Object {
  Symbol(chalk): [Function],
  Symbol(data): Object {
    "chalkOptions": Object {},
    "stream": true,
  },
}
`;

exports[`styleNamesWithoutFn blue 1`] = `"function"`;

exports[`styleNamesWithoutFn blue 2`] = `
Object {
  Symbol(chalk): [Function],
  Symbol(data): Object {
    "chalkOptions": Object {},
    "stream": true,
  },
}
`;

exports[`styleNamesWithoutFn blueBright 1`] = `"function"`;

exports[`styleNamesWithoutFn blueBright 2`] = `
Object {
  Symbol(chalk): [Function],
  Symbol(data): Object {
    "chalkOptions": Object {},
    "stream": true,
  },
}
`;

exports[`styleNamesWithoutFn bold 1`] = `"function"`;

exports[`styleNamesWithoutFn bold 2`] = `
Object {
  Symbol(chalk): [Function],
  Symbol(data): Object {
    "chalkOptions": Object {},
    "stream": true,
  },
}
`;

exports[`styleNamesWithoutFn cmyk 1`] = `"function"`;

exports[`styleNamesWithoutFn cmyk 2`] = `
Object {
  Symbol(chalk): [Function],
  Symbol(data): Object {
    "chalkOptions": Object {},
    "stream": true,
  },
}
`;

exports[`styleNamesWithoutFn cyan 1`] = `"function"`;

exports[`styleNamesWithoutFn cyan 2`] = `
Object {
  Symbol(chalk): [Function],
  Symbol(data): Object {
    "chalkOptions": Object {},
    "stream": true,
  },
}
`;

exports[`styleNamesWithoutFn cyanBright 1`] = `"function"`;

exports[`styleNamesWithoutFn cyanBright 2`] = `
Object {
  Symbol(chalk): [Function],
  Symbol(data): Object {
    "chalkOptions": Object {},
    "stream": true,
  },
}
`;

exports[`styleNamesWithoutFn dim 1`] = `"function"`;

exports[`styleNamesWithoutFn dim 2`] = `
Object {
  Symbol(chalk): [Function],
  Symbol(data): Object {
    "chalkOptions": Object {},
    "stream": true,
  },
}
`;

exports[`styleNamesWithoutFn gray 1`] = `"function"`;

exports[`styleNamesWithoutFn gray 2`] = `
Object {
  Symbol(chalk): [Function],
  Symbol(data): Object {
    "chalkOptions": Object {},
    "stream": true,
  },
}
`;

exports[`styleNamesWithoutFn green 1`] = `"function"`;

exports[`styleNamesWithoutFn green 2`] = `
Object {
  Symbol(chalk): [Function],
  Symbol(data): Object {
    "chalkOptions": Object {},
    "stream": true,
  },
}
`;

exports[`styleNamesWithoutFn greenBright 1`] = `"function"`;

exports[`styleNamesWithoutFn greenBright 2`] = `
Object {
  Symbol(chalk): [Function],
  Symbol(data): Object {
    "chalkOptions": Object {},
    "stream": true,
  },
}
`;

exports[`styleNamesWithoutFn grey 1`] = `"function"`;

exports[`styleNamesWithoutFn grey 2`] = `
Object {
  Symbol(chalk): [Function],
  Symbol(data): Object {
    "chalkOptions": Object {},
    "stream": true,
  },
}
`;

exports[`styleNamesWithoutFn hcg 1`] = `"function"`;

exports[`styleNamesWithoutFn hcg 2`] = `
Object {
  Symbol(chalk): [Function],
  Symbol(data): Object {
    "chalkOptions": Object {},
    "stream": true,
  },
}
`;

exports[`styleNamesWithoutFn hidden 1`] = `"function"`;

exports[`styleNamesWithoutFn hidden 2`] = `
Object {
  Symbol(chalk): [Function],
  Symbol(data): Object {
    "chalkOptions": Object {},
    "stream": true,
  },
}
`;

exports[`styleNamesWithoutFn inverse 1`] = `"function"`;

exports[`styleNamesWithoutFn inverse 2`] = `
Object {
  Symbol(chalk): [Function],
  Symbol(data): Object {
    "chalkOptions": Object {},
    "stream": true,
  },
}
`;

exports[`styleNamesWithoutFn italic 1`] = `"function"`;

exports[`styleNamesWithoutFn italic 2`] = `
Object {
  Symbol(chalk): [Function],
  Symbol(data): Object {
    "chalkOptions": Object {},
    "stream": true,
  },
}
`;

exports[`styleNamesWithoutFn lab 1`] = `"function"`;

exports[`styleNamesWithoutFn lab 2`] = `
Object {
  Symbol(chalk): [Function],
  Symbol(data): Object {
    "chalkOptions": Object {},
    "stream": true,
  },
}
`;

exports[`styleNamesWithoutFn lch 1`] = `"function"`;

exports[`styleNamesWithoutFn lch 2`] = `
Object {
  Symbol(chalk): [Function],
  Symbol(data): Object {
    "chalkOptions": Object {},
    "stream": true,
  },
}
`;

exports[`styleNamesWithoutFn magenta 1`] = `"function"`;

exports[`styleNamesWithoutFn magenta 2`] = `
Object {
  Symbol(chalk): [Function],
  Symbol(data): Object {
    "chalkOptions": Object {},
    "stream": true,
  },
}
`;

exports[`styleNamesWithoutFn magentaBright 1`] = `"function"`;

exports[`styleNamesWithoutFn magentaBright 2`] = `
Object {
  Symbol(chalk): [Function],
  Symbol(data): Object {
    "chalkOptions": Object {},
    "stream": true,
  },
}
`;

exports[`styleNamesWithoutFn red 1`] = `"function"`;

exports[`styleNamesWithoutFn red 2`] = `
Object {
  Symbol(chalk): [Function],
  Symbol(data): Object {
    "chalkOptions": Object {},
    "stream": true,
  },
}
`;

exports[`styleNamesWithoutFn redBright 1`] = `"function"`;

exports[`styleNamesWithoutFn redBright 2`] = `
Object {
  Symbol(chalk): [Function],
  Symbol(data): Object {
    "chalkOptions": Object {},
    "stream": true,
  },
}
`;

exports[`styleNamesWithoutFn reset 1`] = `"function"`;

exports[`styleNamesWithoutFn reset 2`] = `
Object {
  Symbol(chalk): [Function],
  Symbol(data): Object {
    "chalkOptions": Object {},
    "stream": true,
  },
}
`;

exports[`styleNamesWithoutFn strikethrough 1`] = `"function"`;

exports[`styleNamesWithoutFn strikethrough 2`] = `
Object {
  Symbol(chalk): [Function],
  Symbol(data): Object {
    "chalkOptions": Object {},
    "stream": true,
  },
}
`;

exports[`styleNamesWithoutFn underline 1`] = `"function"`;

exports[`styleNamesWithoutFn underline 2`] = `
Object {
  Symbol(chalk): [Function],
  Symbol(data): Object {
    "chalkOptions": Object {},
    "stream": true,
  },
}
`;

exports[`styleNamesWithoutFn visible 1`] = `"function"`;

exports[`styleNamesWithoutFn visible 2`] = `
Object {
  Symbol(chalk): [Function],
  Symbol(data): Object {
    "chalkOptions": Object {},
    "stream": true,
  },
}
`;

exports[`styleNamesWithoutFn white 1`] = `"function"`;

exports[`styleNamesWithoutFn white 2`] = `
Object {
  Symbol(chalk): [Function],
  Symbol(data): Object {
    "chalkOptions": Object {},
    "stream": true,
  },
}
`;

exports[`styleNamesWithoutFn whiteBright 1`] = `"function"`;

exports[`styleNamesWithoutFn whiteBright 2`] = `
Object {
  Symbol(chalk): [Function],
  Symbol(data): Object {
    "chalkOptions": Object {},
    "stream": true,
  },
}
`;

exports[`styleNamesWithoutFn xyz 1`] = `"function"`;

exports[`styleNamesWithoutFn xyz 2`] = `
Object {
  Symbol(chalk): [Function],
  Symbol(data): Object {
    "chalkOptions": Object {},
    "stream": true,
  },
}
`;

exports[`styleNamesWithoutFn yellow 1`] = `"function"`;

exports[`styleNamesWithoutFn yellow 2`] = `
Object {
  Symbol(chalk): [Function],
  Symbol(data): Object {
    "chalkOptions": Object {},
    "stream": true,
  },
}
`;

exports[`styleNamesWithoutFn yellowBright 1`] = `"function"`;

exports[`styleNamesWithoutFn yellowBright 2`] = `
Object {
  Symbol(chalk): [Function],
  Symbol(data): Object {
    "chalkOptions": Object {},
    "stream": true,
  },
}
`;
